### PR TITLE
[APB-1593][LS] Remove whitespace from postCode when sending to backend

### PR DESCRIPTION
**Pull request by Smudgey**

### CODE REVIEW

File: app/uk/gov/hmrc/agentepayeregistrationfrontend/models/Address.scala [hunk 1]
```diff
@@ -23,7 +23,10 @@ case class Address(
   addressLine2: String,
   addressLine3: Option[String],
   addressLine4: Option[String],
-  postCode: String)
+  postCode: String) {
+
+  def withoutSpacesInPostCode: String = postCode.replaceAll(" ", "")
```
Comment (cavster) [inline app/uk/gov/hmrc/agentepayeregistrationfrontend/models/Address.scala]: What is wrong with .trim?

File: app/uk/gov/hmrc/agentepayeregistrationfrontend/controllers/AgentEpayeRegistrationController.scala [hunk 2]
```diff
@@ -86,7 +86,7 @@ class AgentEpayeRegistrationController @Inject() (
       },
       data => {
         Form(single("amend" -> text)).bindFromRequest().fold(
-          _ => registrationService.register(data).map(x => Ok(html.registration_confirmation(x.value))),
+          _ => registrationService.register(withNoSpacesInPostCode(data)).map(x => Ok(html.registration_confirmation(x.value))),
```
Comment (cavster) [inline app/uk/gov/hmrc/agentepayeregistrationfrontend/controllers/AgentEpayeRegistrationController.scala]: Maybe replace x with something more meaningful ;)

Comment (Smudgey) [reply app/uk/gov/hmrc/agentepayeregistrationfrontend/controllers/AgentEpayeRegistrationController.scala]: Feel free to do so

File: app/uk/gov/hmrc/agentepayeregistrationfrontend/models/Address.scala (hunk 1 above)
Comment (Smudgey) [reply app/uk/gov/hmrc/agentepayeregistrationfrontend/models/Address.scala]: It 'trims' the ends, it doesn't replace spaces in the middle

File: app/uk/gov/hmrc/agentepayeregistrationfrontend/controllers/AgentEpayeRegistrationController.scala (hunk 2 above)
Comment (cavster) [reply app/uk/gov/hmrc/agentepayeregistrationfrontend/controllers/AgentEpayeRegistrationController.scala]: Ah so your right !

File: app/uk/gov/hmrc/agentepayeregistrationfrontend/models/Address.scala (hunk 1 above)
Comment (cavster) [reply app/uk/gov/hmrc/agentepayeregistrationfrontend/models/Address.scala]: Correct!

**Pull request merged**